File: app/src/test/java/bj/discogsbrowser/label/LabelFactory.java
package bj.discogsbrowser.label;

import java.util.Arrays;
import java.util.List;

import bj.discogsbrowser.model.label.Label;
import bj.discogsbrowser.model.labelrelease.LabelRelease;
import bj.discogsbrowser.model.release.Image;

/**
 * Created by Josh Laird on 09/05/2017.
 */
public class LabelFactory
{
    public TestLabel getLabel()
    {
        return new TestLabel();
    }

    public List<LabelRelease> getThreeLabelReleases()
    {
        LabelRelease[] labelReleases1 = {new LabelRelease(), new LabelRelease(), new LabelRelease()};
        return Arrays.asList(labelReleases1);
    }

    public List<LabelRelease> getSixLabelReleases()
    {
        LabelRelease[] labelReleases1 = {new LabelRelease(), new LabelRelease(), new LabelRelease(), new LabelRelease(), new LabelRelease(), new LabelRelease()};
        return Arrays.asList(labelReleases1);
    }

    private class TestLabel extends Label
    {
        @Override
        public List<Image> getImages()
        {
            Image image = new Image();
            image.setUri("");
            return Arrays.asList(image);
        }
    }
}

File: app/src/testShared/bj/discogsbrowser/label/LabelFactory.kt
package bj.discogsbrowser.label

import bj.discogsbrowser.model.labelrelease.LabelRelease

/**
 * Created by Josh Laird on 19/05/2017.
 */
object LabelFactory {
    @JvmStatic fun buildNumberOfLabelReleases(number: Int): List<LabelRelease> {
        return (1..number).map { buildLabelRelease(it) }
    }

    @JvmStatic fun buildLabelRelease(number: Int): LabelRelease {
        val labelRelease = LabelRelease()
        labelRelease.title = "labelRelease" + number
        labelRelease.artist = "labelArtist" + number
        labelRelease.catno = "CATNO" + number
        labelRelease.id = "labelReleaseId" + number
        return labelRelease
    }
}